common changes on the lantiq target
- add packages to modules.mk
- disable definiton of SUBTARGETS to trigger automatic scan instead

diff --git a/target/linux/lantiq/Makefile b/target/linux/lantiq/Makefile
--- a/target/linux/lantiq/Makefile
+++ b/target/linux/lantiq/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 BOARD:=lantiq
 BOARDNAME:=Lantiq
 FEATURES:=squashfs
-SUBTARGETS=xway xrx200
+#SUBTARGETS=xway xrx200
 MAINTAINER:=John Crispin <blogic@openwrt.org>
 
 KERNEL_PATCHVER:=3.18
diff --git a/target/linux/lantiq/modules.mk b/target/linux/lantiq/modules.mk
--- a/target/linux/lantiq/modules.mk
+++ b/target/linux/lantiq/modules.mk
@@ -20,3 +20,61 @@ endef
 
 $(eval $(call KernelPackage,i2c-lantiq))
 
+define KernelPackage/lantiq-vpe
+  TITLE:=Lantiq VPE extensions
+  SUBMENU:=Lantiq
+  DEPENDS:=@TARGET_lantiq +kmod-vpe
+  KCONFIG:=CONFIG_IFX_VPE_CACHE_SPLIT=y \
+	  CONFIG_IFX_VPE_EXT=y \
+	  CONFIG_VPE_SOFTDOG=n \
+	  CONFIG_MTSCHED=y \
+	  CONFIG_PERFCTRS=n
+endef
+
+define KernelPackage/lantiq-vpe/description
+  Kernel extensions for the Lantiq SoC
+endef
+
+$(eval $(call KernelPackage,lantiq-vpe))
+
+define KernelPackage/lantiq-nf
+  TITLE:=Lantiq NF extensions
+  SUBMENU:=Lantiq
+  DEPENDS:=@TARGET_lantiq
+  KCONFIG:=CONFIG_NF_CONNTRACK_EVENTS=y
+endef
+
+define KernelPackage/lantiq-nf/description
+  Netfilter extensions for the Lantiq SoC
+endef
+
+$(eval $(call KernelPackage,lantiq-nf))
+
+define KernelPackage/spi-lantiq-ssc
+  SUBMENU:=$(SPI_MENU)
+  TITLE:=Lantiq SPI controller
+  DEPENDS:=@TARGET_lantiq +kmod-spi-bitbang
+  KCONFIG:=CONFIG_SPI_XWAY \
+	  CONFIG_SPI_XWAY_BV=y
+  FILES:=$(LINUX_DIR)/drivers/spi/spi-xway.ko
+endef
+
+define KernelPackage/spi-lantiq-ssc/description
+  Lantiq SPI controller
+endef
+
+$(eval $(call KernelPackage,spi-lantiq-ssc))
+
+define KernelPackage/spi-lantiq-ssc-csi
+  SUBMENU:=$(SPI_MENU)
+  TITLE:=Lantiq SPI controller for CSI
+  DEPENDS:=@TARGET_lantiq +kmod-spi-bitbang
+  KCONFIG:=CONFIG_SPI_XWAY_CSI
+  FILES:=$(LINUX_DIR)/drivers/spi/spi-xway-csi.ko
+endef
+
+define KernelPackage/spi-lantiq-ssc-csi/description
+  Lantiq SPI controller for CSI
+endef
+
+$(eval $(call KernelPackage,spi-lantiq-ssc-csi))
